### ISSUE TITLE
Terraform 0.13 registry module syntax appears to be unsupported

### ISSUE THREAD

**Issue by mike-morr** [gruntwork-io/terragrunt#1380]
```hcl
# Terragrunt will copy the Terraform configurations specified by the source parameter, along with any files in the
# working directory, into a temporary folder, and execute your Terraform commands in that folder.
terraform {
  source  = "terraform-aws-modules/atlantis/aws"
  version = "2.24.0"
}
```

I am trying to use the new module syntax shown above in TF 0.13 on TG 0.25.2 and getting the following error:

```
project/us-west-2/dev/atlantis_fargate/terragrunt.hcl:22,3-10: Unsupported argument; An argument named "version" is not expected here.
```

I am assuming this is just not supported yet.  If so, this is a feature request.  Otherwise, it is a support request.  :-)

Thank you for the amazing project!  🥇

**Comment by mike-morr** [comment]
FYI.  I tried adding the ```version``` to the inputs as a workaround, but it resulted in:

```
[terragrunt] 2020/10/06 22:07:18 WARNING: no double-slash (//) found in source URL project/us-west-2/dev/atlantis_fargate/terraform-aws-modules/atlantis/aws. Relative paths in downloaded Terraform code may not work.
```

**Comment by brikis98** [comment]
This is not something currently supported. Right now, Terragrunt solely supports the `source` param. PRs to support an optional, separate `version` param are welcome.

**Comment by kitos9112** [comment]
Hmm - Trying to understand what's the actual limitation here. Could you not just use `locals` to perform an in-place interpolation following the standard `git` module source of Terraform?

`
# project/us-west-2/dev/atlantis_fargate/terragrunt.hcl
locals {
  module_url = "git::https://github.com/hashicorp/terraform-aws-consul.git" # << This could come from anywhere else in your TG project
  module_version = "v0.0.8" << This could come from anywhere else in your TG project
}

terraform {
  source = "${local.module_url}?ref=${local.module_version}"
}

`

**Comment by yhakbar** [comment]
@kitos9112 that doesn't work if you are utilizing terraform.io or another registry provider to download the modules instead of git.

See https://www.terraform.io/docs/registry/modules/use.html#using-modules

**Comment by kitos9112** [comment]
Fair point - not consuming much Terraform content directly from the registry. You're right, this could be an enhancement to Terragrunt :)

**Comment by yhakbar** [comment]
@kitos9112 is there a formal way I can make this feature request? I would try to provide a PR to address this issue, but I don't think I'm familiar enough with this project or competent enough with Golang to do so.

**Comment by yorinasub17** [comment]
> is there a formal way I can make this feature request?

This is the ticket for the feature request (the `enhancement` tag). We're buried right now though with other initiatives so unlikely that we (Gruntwork) will get to this anytime soon. If anyone wants to contribute this from the community, the key places to update are:

- `config` needs to be updated to parse the new `version` attribute
- `cli/tfsource` needs to be updated to handle the registry syntax for downloading the source
- `cli` needs to be updated to translate the `config` to `cli/tfsource`
